app/assets/javascripts/discourse/app/components/glimmer-header/icons.gjs

```diff
@@ -23,12 +23,21 @@ export default class Icons extends Component {
   @service header;
   @service search;
 
+  _isStringType = (icon) => typeof icon === "string";
+
   <template>
     <ul class="icons d-header-icons">
-      {{#each _extraHeaderIcons as |icon|}}
-        <MountWidget @widget={{icon}} />
-        {{! I am not sure how we are going to render glimmer components here without
-        being able to import them. }}
+      {{#each _extraHeaderIcons as |Icon|}}
+        {{#if (this._isStringType Icon)}}
+          <MountWidget @widget={{Icon}} />
+        {{else}}
+          {{#let
+            (component PanelWrapper panelElement=@panelElement)
```

Comment:
Could also do

```
(component ConditionalInElement element=@panelElement)
```
if we want to avoid the extra `PanelWrapper` class.

(if you wanna keep the separate wrapper then that's also fine - I guess it comes down to personal preference)

Reply:
I think I prefer the consistency. I will leave it for now.